gematria/granite/graph_builder_model_inference.cc

```diff
@@ -428,25 +424,31 @@ GraphBuilderModelInference::FromTfLiteModel(
 
   // Get a mapping between the input tensors used in the model and their
   // corresponding indices.
-  auto input_tensor_to_idx =
-      std::make_unique<std::vector<int>>(kNumPossibleInputTensors, -1);
+  std::vector<int> input_tensor_to_idx(kNumPossibleInputTensors, -1);
   for (int idx : (*interpreter)->inputs()) {
     const std::string_view name = (*interpreter)->GetInputName(idx);
-    input_tensor_to_idx->at(static_cast<int>(GetInputTensorFromName(name))) =
-        idx;
+    std::optional<InputTensor> input_tensor = GetInputTensorFromName(name);
+
+    // Report any unexpected tensors.
+    if (!input_tensor.has_value()) {
+      llvm::errs() << "Model contains unexpected tensor `" << name
+                   << "` at index " << idx << '\n';
+      continue;
```

Comment:
We should probably return an error here (or collect all the unexpected tensor names and then return them all in an error, that's even more useful), because it means that we won't be able to use this model in the inference library (there's no code that would fill in the unknown input tensor).

As a side note: in library code, it's usually better to keep logging light or do it only when requested (and return important information through the status codes, etc.). Otherwise, this quickly makes the logs very hard to read without additional tools for filtering.